File: app/src/main/java/org/metabrainz/mobile/presentation/features/UserPreferences.java
package org.metabrainz.mobile.presentation.features;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import org.metabrainz.mobile.App;

public class UserPreferences {

    public static final String PREFERENCE_GET_PRIVATE_COLLECTIONS = "private_collections";

    public static SharedPreferences getPreferences() {
        return PreferenceManager.getDefaultSharedPreferences(App.getContext());
    }

    public static void setPreferenceGetPrivateCollections(boolean choice) {
        SharedPreferences.Editor editor = UserPreferences.getPreferences().edit();
        editor.putBoolean(PREFERENCE_GET_PRIVATE_COLLECTIONS, choice);
        editor.apply();
    }

    public static boolean getPrivateCollectionsPreference() {
        return UserPreferences.getPreferences().getBoolean(PREFERENCE_GET_PRIVATE_COLLECTIONS, false);
    }

}

File: app/src/main/java/org/metabrainz/android/presentation/UserPreferences.kt
package org.metabrainz.android.presentation

import androidx.preference.PreferenceManager
import org.metabrainz.android.App
import org.metabrainz.android.presentation.features.login.LoginSharedPreferences

object UserPreferences {

    const val PREFERENCE_CLEAR_SUGGESTIONS = "clear_suggestions"
    const val PREFERENCE_LISTENBRAINZ_TOKEN = "listenbrainz_user_token"
    const val PREFERENCE_PICARD_PORT = "picard_port"
    const val PREFERENCE_IP_ADDRESS = "ip_address"
    const val PREFERENCE_LISTENING_ENABLED = "listening_enabled"
    private const val PREFERENCE_LISTENING_SPOTIFY = "listening_spotify_enabled"
    private const val PREFERENCE_GET_PRIVATE_COLLECTIONS = "private_collections"
    private const val PREFERENCE_RATINGS_TAGS = "ratings_tags"
    private const val PREFERENCE_SYSTEM_LANGUAGE = "use_english"
    const val PREFERENCE_SYSTEM_THEME = "app_theme"
    private const val PREFERENCE_ONBOARDING = "onboarding"
    private val preferences = PreferenceManager.getDefaultSharedPreferences(App.context!!)

    fun setOnBoardingCompleted() {
        val editor = preferences.edit()
        editor.putBoolean(PREFERENCE_ONBOARDING, true)
        editor.apply()
    }

    val privateCollectionsPreference = preferences.getBoolean(PREFERENCE_GET_PRIVATE_COLLECTIONS, false)
    val ratingsTagsPreference = preferences.getBoolean(PREFERENCE_RATINGS_TAGS, false)
    val systemLanguagePreference = preferences.getBoolean(PREFERENCE_SYSTEM_LANGUAGE, false)
    val themePreference = preferences.getString(PREFERENCE_SYSTEM_THEME, "Use device theme")
    
    val preferencePicardPort: String
        get() = preferences.getString(PREFERENCE_PICARD_PORT, "8000")!!
    val preferenceIpAddress: String?
        get() = preferences.getString(PREFERENCE_IP_ADDRESS, null)

    val preferenceListenBrainzToken = preferences.getString(PREFERENCE_LISTENBRAINZ_TOKEN, null)
    var preferenceListeningEnabled: Boolean
        get() = preferences.getBoolean(PREFERENCE_LISTENING_ENABLED, false)
        set(value) {
            val editor = preferences.edit()
            editor.putBoolean(PREFERENCE_LISTENING_ENABLED, value)
            editor.apply()
        }
    val preferenceListeningSpotifyEnabled = preferences.getBoolean(PREFERENCE_LISTENING_SPOTIFY, false)
}